kotlin/src/main/kotlin/org/jdbi/v3/core/kotlin/KotlinMapper.kt

```diff
@@ -38,22 +39,49 @@ import kotlin.reflect.jvm.jvmErasure
 
 private val nullValueRowMapper = RowMapper<Any?> { rs, ctx -> null }
 
-class KotlinMapper(clazz: Class<*>, private val prefix: String = "") : RowMapper<Any> {
-    private val kClass: KClass<*> = clazz.kotlin
+class KotlinMapper<T: Any>(clazz: Class<T>, private val prefix: String = "") : RowMapper<T> {
+
+    companion object {
+
+        /**
+         * Returns a mapper factory that maps to the given kotlin class
+         *
+         * @param type the mapped class
+         * @return a mapper factory that maps to the given kotlin class
+         */
+        fun factory(type: Class<*>): RowMapperFactory {
+            return RowMapperFactory.of(type, KotlinMapper(type))
+        }
+
+        /**
+         * Returns a mapper factory that maps to the given kotlin class
+         *
+         * @param type the mapped class
+         * @param prefix the column name prefix for each mapped kotlin property
+         * @return a mapper factory that maps to the given kotlin class
+         */
+        fun factory(type: Class<*>, prefix: String): RowMapperFactory {
+            return RowMapperFactory.of(type, KotlinMapper(type, prefix))
+        }
+
+
+    }
+
+    private val kClass = clazz.kotlin
     private val constructor = findConstructor(kClass)
     private val constructorParameters = constructor.parameters
     private val memberProperties = kClass.memberProperties.mapNotNull { it as? KMutableProperty1<*, *>}.filter { property ->
         !constructorParameters.any { parameter -> parameter.paramName() == property.propName() }
     }
 
-    private val nestedMappers = ConcurrentHashMap<KParameter, KotlinMapper>()
-    private val nestedPropertyMappers = ConcurrentHashMap<KMutableProperty1<*, *>, KotlinMapper>()
+    private val nestedMappers = ConcurrentHashMap<KParameter, KotlinMapper<*>>()
+    private val nestedPropertyMappers = ConcurrentHashMap<KMutableProperty1<*, *>, KotlinMapper<*>>()
```

Comment:
If these lines do not compile without the `<*>` type parameter, then I think it's safe to assume that adding the type parameter is unsafe in Kotlin